src/stk/io/nifti.cpp

```diff
@@ -6,6 +6,56 @@
 
 // TODO: Byte swapping?
 
+/** Transpose vector data to have the vector index in the slowest axis */
+template<typename T>
+void to_slowest(
+        const T * const __restrict in,
+        T * const __restrict out,
+        const dim3 size,
+        const int nt
+        )
+{
+    const int nx = (int) size.x;
+    const int ny = (int) size.y;
+    const int nz = (int) size.z;
+
+    #pragma omp parallel for
+    for (int z = 0; z < nz; ++z) {
+        for (int y = 0; y < ny; ++y) {
+            for (int x = 0; x < nx; ++x) {
+                for (int t = 0; t < nt; ++t) {
+                    out[t*nz*ny*nx + z*ny*nx + y*nx + x] = in[z*ny*nx*nt + y*nx*nt + x*nt + t];
```

Comment:
This could probably made faster using intrinsics, but I don't think the gain here is worth bothering.